htdocs/core/lib/functions.lib.php

```diff
@@ -14016,6 +14016,11 @@ function dolForgeCriteriaCallback($matches)
 		}
 	}
 
+	// Filter on DATE(t.field) if $tmpescaped equal 'YYYY-MM-DD'
+	if (strpos($operand, 'date') !== false && preg_match("/^'\d{4}\-\d{2}\-\d{2}'\$/", $tmpescaped)) {
+		$operand = 'DATE('.$operand.')';
```

Comment:
Use of DATE function is forbidden. See https://wiki.dolibarr.org/index.php?title=Language_and_development_rules#SQL_Coding_rules

Try to use sqlDateFilter() instead.